spyder/api/widgets/auxiliary_widgets.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Spyder API auxiliary widgets.
+"""
+
+# Third party imports
+from qtpy.QtWidgets import QHBoxLayout, QMainWindow, QWidget
+import qdarkstyle
+
+# Local imports
+from spyder.api.exceptions import SpyderAPIError
+from spyder.config.gui import is_dark_interface
+
+
+class SpyderWindowWidget(QMainWindow):
+    """
+    MainWindow subclass that contains a Spyder Plugin.
+    """
+
+    def __init__(self, widget):
+        super().__init__()
+        self.widget = widget
+
+        # Setting interface theme
+        if is_dark_interface():
+            self.setStyleSheet(qdarkstyle.load_stylesheet())
+
+    def closeEvent(self, event):
+        """
+        Override Qt method.
+        """
+        # TODO:
```

Comment:
Do you plan to address this TODO before merging this?

Reply:
Fixed